### PR TITLE
drivers: clock_control: stm32h5: Add MCO sources and dividers

### PR DESCRIPTION
**Pull request by Dalachowsky**
STM32H5 series lacked support for MCO configuration. Added SOC_SERIES_STM32H5X to approperiate kconfig MCO source configurations. Added new MCO sources from H5 series and updated the clock_stm32_ll_h5.c with MCO configuration.

### CODE REVIEW

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: drivers/clock_control/Kconfig.stm32
Comment (Dalachowsky) [reply drivers/clock_control/Kconfig.stm32:167]: That's right I'll fix it in a moment. Should i just specify the range like:
```
range 1 5  if !SOC_SERIES_STM32H5X
range 1 15 if SOC_SERIES_STM32H5X
```

Comment (FRASTM) [reply drivers/clock_control/Kconfig.stm32:167]: yes, or like  this is done for _LL_RCC_MCO1_DIV_ :

```
	range 1   5 if SOC_SERIES_STM32F4X || SOC_SERIES_STM32F7X
	range 1   15 if SOC_SERIES_STM32H5X
```

- and also  change the description is `help` section too,

(Moreover the range 1..5 also exists for SOC_SERIES_STM32F2X and range 1..15 also exists for SOC_SERIES_STM32H7X, SOC_SERIES_STM32MP1X but that's not the purpose here)

**Comment by faloj** [comment]
This PR will conflict with #60448 wich add MCO support for the H7 family but also do some code factorization to avoid  stm32_clock_control_mco_init duplication. Maybe wait for it to be merged before merging this one to benefit from the code factorization ?

**Comment by Dalachowsky** [comment]
@faloj Thanks for mentioning your PR.
I was also considering extracting mco to separate common file so I'll wait until yours is accepted.

**Comment by faloj** [comment]
PR #60448 has been merged ! You can integrate the changements into your work ;)

File: drivers/clock_control/Kconfig.stm32
Comment (erwango) [inline drivers/clock_control/Kconfig.stm32:70]: Please fix indentation.
Apply everywhere

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/clock_control/Kconfig.stm32
Comment (faloj) [inline drivers/clock_control/Kconfig.stm32:228]: This clock source already exist and is named "CLOCK_STM32_MCO2_SRC_PLLPCLK"

File: drivers/clock_control/Kconfig.stm32
Comment (faloj) [inline drivers/clock_control/Kconfig.stm32:234]: Same here with "CLOCK_STM32_MCO2_SRC_PLL2PCLK"

**Review by faloj — CHANGES_REQUESTED** [review]

File: drivers/clock_control/Kconfig.stm32
Comment (Dalachowsky) [reply drivers/clock_control/Kconfig.stm32]: I have fixed the indentation and applied suggestions of @faloj
Sorry for making a bit of mess in this PR

File: drivers/clock_control/Kconfig.stm32
Comment (faloj) [reply drivers/clock_control/Kconfig.stm32:228]: This should be removed now that you added "CLOCK_STM32_MCO2_SRC_PLLPCLK" for SOC_SERIES_STM32H5 ;)

**Review by faloj — CHANGES_REQUESTED** [review]

**Review by FRASTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**
